pkg/podreadiness/conditioninjector.go

```diff
@@ -0,0 +1,67 @@
+package podreadiness
+
+import (
+	"context"
+	"fmt"
+	"os"
+
+	v1 "k8s.io/api/core/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/client-go/kubernetes"
+	"k8s.io/klog/v2"
+)
+
+type ConditionInjector struct {
+	cs      *kubernetes.Clientset
+	ns      string
+	podName string
+}
+
+func NewConditionInjector(cs *kubernetes.Clientset) (*ConditionInjector, error) {
+	nsVal, ok := os.LookupEnv("REFERENCE_NAMESPACE")
```

Comment:
the naming is a bit unusual, I often saw `MY_POD_NAMESPACE` (and `MY_POD_NAME`) probably because almost everyone lazily mirrors the docs/code examples.

Reply:
I didn't write it. it was already defined under the DaemonSet definition.

Reply:
Right. Then, it's good as it is now.